extensions/vscode/src/extension.ts

```diff
@@ -22,3 +27,51 @@ export function activate(
   );
   return context;
 }
+
+/**
+ * Checks if the version of Dart Frog CLI installed in the user's system is
+ * compatible with this extension, suggesting to install if it is not.
+ *
+ * This method should be called upon activation of the extension to ensure that
+ * the version of Dart Frog CLI installed in the user's system is compatible
+ * with this extension.
+ *
+ * If the version of Dart Frog CLI installed in the user's system is not
+ * compatible with this extension, the user is prompted to update Dart Frog CLI
+ * or ignore the warning. If the user chooses to update Dart Frog CLI, the
+ * extension will attempt to update Dart Frog CLI. Otherwise, the user will
+ * proceed to use the extension at their own risk.
+ *
+ * If Dart Frog CLI is not installed in the user's system, this method will
+ * do nothing and returns immediately.
+ *
+ * @see {@link isCompatibleDartFrogCLIVersion}, to check if the version of
+ * Dart Frog CLI installed in the user's system is compatible with this
+ * extension.
+ */
+export async function ensureCompatibleDartFrogCLI(): Promise<void> {
+  const version = readDartFrogCLIVersion();
+  if (!version) {
+    return;
+  }
+
+  const versionIsCompatible = isCompatibleDartFrogCLIVersion(version);
+  if (versionIsCompatible) {
+    return;
+  }
+
+  const selection = await vscode.window.showWarningMessage(
+    `Dart Frog CLI version ${version} is not compatible with this extension.`,
+    "Update Dart Frog CLI",
```

Comment:
And what if the version is newer yet not compatible? Should we maybe check for that as well

Reply:
That's a valid scenario. Currently I'm assuming we're keeping the extension up to date with the latest Dart Frog CLI version.

I think it will be weird if we didn't keep it up to date and forced users to rollback.

We can implement an option to allow the users rolling-back their versions, but ideally in a different Pull Request to avoid pullting this one. What do you think? If you agree, we can open an issue to keep track of this suggested feature 💙

Reply:
The warning is enough, so lets keep it as is.